File: domain/src/main/java/com/insurtech/kanguro/domain/model/CloudDocument.kt
package com.insurtech.kanguro.domain.model

import java.util.Date

data class CloudDocument(
    val userId: String? = null,
    val pets: List<CloudPet>? = null,
    val renters: List<CloudRenters>? = null
)

data class CloudPet(
    val id: Long? = null,
    val name: String? = null,
    val cloudDocumentPolicies: List<CloudDocumentPolicy>? = null
)

data class CloudDocumentPolicy(
    val id: String? = null,
    val ciId: Long? = null,
    val policyStartDate: Date? = null,
    val policyAttachments: List<PolicyAttachment>? = null,
    val policyDocuments: List<PolicyDocument>? = null,
    val claimDocuments: List<CloudClaimDocument>? = null
)

data class CloudClaimDocument(
    val claimId: String? = null,
    val claimPrefixId: String? = null,
    val claimDocuments: List<ClaimDocument>? = null
)

File: iosApp/Modules/KanguroSharedDomain/Sources/KanguroSharedDomain/CloudDocument/Model/CloudDocument.swift
import Foundation

public struct CloudDocument: Equatable {

    public let userId: String?
    public let pets: [PetCloud]?
    public let renters: [RentersCloud]?

    public init(userId: String?, pets: [PetCloud]?, renters: [RentersCloud]?) {
        self.userId = userId
        self.pets = pets
        self.renters = renters
    }
}
